### DISCUSSION TITLE
DB size estimation for Europe/Italy

### DISCUSSION THREAD

**Discussion by KylesDev** [osm-search/Nominatim#2990] [Q&A] [answered]
Hello, i want to set up an instance of Nominatim with the full data from Europe, or at least just for Italy, but i need to make an estimation of the hardware requirements (especially RAM and HDD/SSD Size) for the machine that will be used to deploy it; 
i've checked the official docs, but they provide only estimations for the full planet installation, does anyone know some other way to calculate the necessary space and RAM? 
Thanks in advice

**Comment by lonvia** [comment, marked as answer]
As a rule-of-thumb you can take the size of the PBF for the planet and the size of the PBF for your extract, do a simple linear interpolation of RAM and disk size and round up a bit. So for Europe and Italy something like that:

|         | PBF size | % of planet | RAM   | disk space |
| ------- | -------- | ----------- | ----- | ---------- |
| planet  | 68GB     | 100%        | 128GB | 900GB      |
| europe  | 27GB     | 40%         | 64GB  | 400GB      |
| italy   | 2GB      | 3%          | 4GB   | 35GB       |

RAM is always a lower bound. Having more RAM never hurts. For disk space add a fixed 80GB for the flat node file if you use one (strongly recommended for planet and europe). If you want to run updates, add 10% disk space for index bloat and on top of that 10% database growth per year.
